diff -up onnx-1.19.0/CMakeLists.txt.1~ onnx-1.19.0/CMakeLists.txt
--- onnx-1.19.0/CMakeLists.txt.1~	2025-08-26 23:37:09.000000000 +0200
+++ onnx-1.19.0/CMakeLists.txt	2025-10-06 02:02:54.193264335 +0200
@@ -409,7 +409,7 @@ relative_protobuf_generate_cpp(ONNX_PROT
                                onnx/onnx-operators.in.proto
                                onnx/onnx-data.in.proto)
 
-add_library(onnx_proto ${ONNX_PROTO_SRCS})
+add_library(onnx_proto SHARED ${ONNX_PROTO_SRCS})
 
 file(GLOB_RECURSE __tmp_srcs "${ONNX_ROOT}/onnx/*.h" "${ONNX_ROOT}/onnx/*.cc")
 file(GLOB_RECURSE onnx_gtests_src "${ONNX_ROOT}/onnx/test/cpp/*.h"
@@ -423,6 +423,8 @@ list(APPEND ONNX_SRCS ${__tmp_srcs})
 # Hide all symbols we don't need
 set_target_properties(onnx_proto PROPERTIES CXX_VISIBILITY_PRESET hidden)
 set_target_properties(onnx_proto PROPERTIES VISIBILITY_INLINES_HIDDEN 1)
+set_target_properties(onnx_proto PROPERTIES VERSION ${ONNX_VERSION}
+                                            SOVERSION ${ONNX_VERSION})
 
 set(LINKED_PROTOBUF_TARGET protobuf::libprotobuf)
 if(ONNX_USE_LITE_PROTO)
@@ -459,7 +461,9 @@ if(CMAKE_SYSTEM_NAME STREQUAL "AIX")
   # So, create a object library
   add_library(onnx OBJECT ${ONNX_SRCS})
 else()
-  add_library(onnx ${ONNX_SRCS})
+  add_library(onnx SHARED ${ONNX_SRCS})
+  set_target_properties(onnx PROPERTIES VERSION ${ONNX_VERSION}
+                                        SOVERSION ${ONNX_VERSION})
 endif()
 set_target_properties(onnx PROPERTIES CXX_VISIBILITY_PRESET hidden)
 set_target_properties(onnx PROPERTIES VISIBILITY_INLINES_HIDDEN ON)
@@ -538,6 +542,17 @@ install(DIRECTORY ${CMAKE_CURRENT_BINARY
         FILES_MATCHING
         PATTERN "*.h")
 
+install(DIRECTORY ${ONNX_ROOT}/onnx
+        DESTINATION "${PY_SITEARCH}"
+        FILES_MATCHING
+        PATTERN "*.py"
+        PATTERN "test/*" EXCLUDE )
+install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/onnx
+        DESTINATION  "${PY_SITEARCH}"
+        FILES_MATCHING
+        PATTERN "*.py"
+        PATTERN "test/*" EXCLUDE )
+
 configure_file(
   ${PROJECT_SOURCE_DIR}/cmake/ONNXConfigVersion.cmake.in
   ${PROJECT_BINARY_DIR}/ONNXConfigVersion.cmake
@@ -590,6 +605,10 @@ install(TARGETS
   onnx onnx_proto
   EXPORT ONNXTargets DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
+install(TARGETS
+  onnx_cpp2py_export
+  DESTINATION "${PY_SITEARCH}/onnx")
+
 install(EXPORT ONNXTargets
   DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/ONNX"
   NAMESPACE ONNX::
