diff -up onnx-1.18.0/CMakeLists.txt.4~ onnx-1.18.0/CMakeLists.txt
--- onnx-1.18.0/CMakeLists.txt.4~	2025-06-09 18:39:12.369618144 +0200
+++ onnx-1.18.0/CMakeLists.txt	2025-06-09 18:39:39.967767360 +0200
@@ -503,15 +503,15 @@ target_compile_features(onnx_proto PUBLI
 
 if(ONNX_USE_LITE_PROTO)
   if(TARGET protobuf::libprotobuf-lite)
-    target_link_libraries(onnx_proto PUBLIC protobuf::libprotobuf-lite PRIVATE ${protobuf_ABSL_USED_TARGETS})
+    target_link_libraries(onnx_proto PUBLIC protobuf::libprotobuf-lite PRIVATE ${protobuf_ABSL_USED_TARGETS} -labsl_log_internal_check_op -labsl_log_internal_message -labsl_log_internal_nullguard)
   else()
-    target_link_libraries(onnx_proto PUBLIC ${PROTOBUF_LITE_LIBRARIES})
+    target_link_libraries(onnx_proto PUBLIC ${PROTOBUF_LITE_LIBRARIES} -labsl_log_internal_check_op -labsl_log_internal_message -labsl_log_internal_nullguard)
   endif()
 else()
   if(TARGET protobuf::libprotobuf)
-    target_link_libraries(onnx_proto PUBLIC protobuf::libprotobuf PRIVATE ${protobuf_ABSL_USED_TARGETS})
+    target_link_libraries(onnx_proto PUBLIC protobuf::libprotobuf PRIVATE ${protobuf_ABSL_USED_TARGETS} -labsl_log_internal_check_op -labsl_log_internal_message -labsl_log_internal_nullguard)
   else()
-    target_link_libraries(onnx_proto PUBLIC ${PROTOBUF_LIBRARIES})
+    target_link_libraries(onnx_proto PUBLIC ${PROTOBUF_LIBRARIES} -labsl_log_internal_check_op -labsl_log_internal_message -labsl_log_internal_nullguard)
   endif()
 endif()
 add_onnx_global_defines(onnx_proto)
@@ -532,7 +532,7 @@ target_include_directories(onnx PUBLIC
   $<BUILD_INTERFACE:${ONNX_ROOT}>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
   $<INSTALL_INTERFACE:include>)
-target_link_libraries(onnx PUBLIC onnx_proto)
+target_link_libraries(onnx PUBLIC onnx_proto -labsl_log_internal_check_op -labsl_log_internal_message -labsl_log_internal_nullguard)
 add_onnx_global_defines(onnx)
 
 if(ONNX_BUILD_PYTHON)
